File: src/main/java/net/sf/rails/common/ResourceLoader.java
package net.sf.rails.common;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.InputStream;

public final class ResourceLoader {

    private static final Logger log = LoggerFactory.getLogger(ResourceLoader.class);

    public static final String SEPARATOR = "/";

    public static InputStream getInputStream(String filename, String directory) {
        String fullPath = directory + SEPARATOR + fixFilename(filename);
        log.trace("Locate fullPath (updated) = {}", fullPath);
        return ResourceLoader.class.getClassLoader().getResourceAsStream(fullPath);
    }

    /**
     * Fix a filename by replacing space with underscore.
     *
     * @param filename Filename to fix.
     * @return The fixed filename.
     */
    private static String fixFilename(String filename) {
        return filename.replace(' ', '_');
    }
}

File: dotnet/RailsLib.Net/Net/Common/ResourceLoader.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GameLib.Net.Common
{
    public sealed class ResourceLoader
    {
        private static Logger<ResourceLoader> log = new Logger<ResourceLoader>();

    public const string SEPARATOR = "/";

        // #FIXME file handling
    //public static InputStream getInputStream(String filename, String directory)
    //    {
    //        String fullPath = directory + SEPARATOR + fixFilename(filename);
    //        log.debug("Locate fullPath (updated) =" + fullPath);
    //        return ResourceLoader.class.getClassLoader().getResourceAsStream(fullPath);
    //}

    /**
     * Fix a filename by replacing space with underscore.
     *
     * @param filename Filename to fix.
     * @return The fixed filename.
     */
    private static string FixFilename(string filename)
    {
        return filename.Replace(' ', '_');
    }
}
}
